apps/server/src/tasks/tasks.service.ts

```diff
@@ -19,6 +20,79 @@ import { Task } from './entities/task.entity';
 
 @Injectable()
 export class TasksService {
+  @Cron('* */1 * * * *')
+  triggerMarkingAsDone() {
+    this.markTasksAsDone();
+  }
+
+  async markTasksAsDone() {
+    const tasks: Task[] = await Task.find({
+      relations: ['chunks', 'chunkInfo'],
+    });
+
+    tasks
+      .filter((task) => task.shouldAutoResolve)
+      .map((task) => {
+        if (task.isFloat) {
+          if (this.isPastDeadline(task)) {
+            // mark task as done (and it's corresponding chunks)
+            Task.update(task.id, {
+              ...task,
+              isDone: true,
+            });
+            task.chunks.map((chunk) => {
+              Chunk.update(chunk.id, {
+                ...chunk,
+                isDone: true,
+              });
+            });
+          } else {
+            // mark specific chunks as done
+            task.chunks.map((chunk) => {
+              Chunk.update(chunk.id, {
+                ...chunk,
+                isDone: this.isChunkEnded(chunk),
+              });
+            });
+          }
+        } else if (task.chunkInfo.repeat) {
+          // mark specific chunks as done
+          task.chunks.map((chunk) => {
+            Chunk.update(chunk.id, {
+              ...chunk,
+              isDone: this.isChunkEnded(chunk),
+            });
+          });
+        } else if (this.isConstTaskEnded(task)) {
+          // mark task as done (and it's corresponding chunks)
+          Task.update(task.id, {
+            ...task,
+            isDone: true,
+          });
```

Comment:
I think there is a better way to update just a single field without the need to pass all the task. (Because now it will do update for ALL fields, even if we're updating one). That's the traffic that we don't need and RAM that we're limited with.